src/fromager/sources.py

```diff
@@ -326,17 +336,36 @@ def _apply_patch(patch: pathlib.Path, source_root_dir: pathlib.Path):
 
 
 def _warn_for_old_patch(
-    source_root_dir: pathlib.Path, patch_filename: pathlib.Path
+    source_root_dir: pathlib.Path,
+    patches_dir: pathlib.Path,
 ) -> None:
-    # Get the existing version and req name as per the source_root_dir naming conventions
+    # Get the req name as per the source_root_dir naming conventions
     req_name = source_root_dir.name.rsplit("-", 1)[0]
-    existing_version = Version(source_root_dir.name.rsplit("-", 1)[1])
 
-    # Check to see if patches for older version exists
-    if patch_filename.parent.name == source_root_dir.name:
-        logger.warning(
-            f"{req_name}: patches for version {existing_version!s} exist but will not be applied"
+    # Get all the patches for that requirement
+    patches = overrides.get_patches(patches_dir, source_root_dir.name)
+
+    if len(patches) > 0:
+        # Filter the patches using regex
+        filtered_patches = _filter_patches(patches, req_name)
+        current_path = source_root_dir.parent.joinpath(
+            "patches_dir", source_root_dir.name
         )
+        # If filtered patches contain patches for current version, remove the path because we dont want to warn
+        if current_path in filtered_patches:
+            filtered_patches.remove(current_path)
+        if len(filtered_patches) > 0:
+            for p in filtered_patches:
+                logger.warning(f"{req_name}: patch {p} exists but will not be applied")
+
+
+# Helper method to filter the unwanted patches using a regex
+def _filter_patches(patches: list[pathlib.Path], req_name: str) -> list[pathlib.Path]:
```

Comment:
This function needs a more descriptive name. 

Maybe this filtering should be done in the function in overrides?

Reply:
Done. Also shifted it to overrides.py